lib/timers/promises.js

```diff
@@ -111,7 +111,86 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options',
+      'Object',
+      options);
+  }
```

Comment:
You can use the util function from `internal/validators`.
```suggestion
  validateObject(options, 'options');
```